assistant-agent/src/agent/middleware/empty_result_guard.py

```
@@ -0,0 +1,190 @@
+            "not found",
+        )
+    ):
+        return True
+    # Length fallback: empty observer responses come back at ~250 bytes
+    # (status + empty-data envelope + metadata). Real responses with even a
+    # single matching log/alert/metric series exceed 400 bytes. Only apply
+    # this check below the conservative 400-byte cutoff to keep the false-
+    # positive rate near zero.
+    if len(text) <= 400:
+        return True
+    return False
```

Ruff (0.15.12)

[warning] Return the condition `len(text) <= 400` directly

Replace with `return len(text) <= 400`

(SIM103)